cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/GlobalRemoteAuthenticatorProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.internal.remote;
+
+import io.cdap.cdap.security.spi.authenticator.RemoteAuthenticator;
+
+import javax.inject.Provider;
+
+/**
+ * Represents a singleton which provides a remote authenticator.
+ */
+public interface GlobalRemoteAuthenticatorProvider extends Provider<RemoteAuthenticator> {
```

Comment:
Why it needs `Global` in the name?

Reply:
Wanted to make it clear this interface is intended to be a singleton.

Reply:
Interface has been removed; see below.